### ISSUE TITLE
Centos 7.2 - Error in configuration after rebooting

### ISSUE THREAD

**Issue by amitwein** [NagiosEnterprises/nagioscore#153]
I have recently installed Nagios 4.2 on a test environment and i've noticed a weird issue.
After rebooting the system, Nagios is not running and i'm getting this error:

Error in configuration file '/usr/local/nagios/etc/nagios.cfg' - Line 452 (Check result path '/usr/local/nagios/var/spool/checkresults' is not a valid directory)

Seems like spool/checkresults disappear after rebooting the system. 

after recreating the spool/checkresults, im getting another error( from nagios.log file) :
qh: Failed to init socket '/usr/local/nagios/var/rw/nagios.qh'. bind() failed: No such file or directory

looks like /usr/local/nagios/var/rw folder also disappeared.
after recreating /rw folder, Nagios is up and running

worth to mention that besides Nagios, there are more components installed like:
Check-mk-livestatus 1.2.8p5 - relay on /usr/local/nagios/var/rw folder
NRPE V3. 0
XINETD
PNP4NAGIOS

**Comment by box293** [comment]
Did you follow any specific guides to install Core ?

Is the location /usr/local/nagios/var a ramdisk by any chance? Or a network share / mount? Wondering if Nagios is starting before the location becomes available.

**Comment by amitwein** [comment]
Hi, 
Nagios is installed on a regular primary partition,
Nagios is starting automatically of course and  the folder /usr/local/nagios/var/rw is disappearing after a reboot and i just can't figure our why

**Comment by ericloyd** [comment]
This sounds like a support issue, because I'm guessing it's specific to your installation, not something endemic to Nagios.  However, some additional information would help:

What Operating System and version?
Physical or virtual?
Can you paste the output of "cat /etc/fstab" here for us to look at?
Can you paste the output of "cat /etc/rc.local" here for us to look at?

**Comment by amitwein** [comment]
Hi, 
that might be related to my extra components.
I'm using a VM.
running on CentOS Linux release 7.2.1511

**fstab:**
#
# /etc/fstab
# Created by anaconda on Mon Jul 13 15:54:26 2015
#
# Accessible filesystems, by reference, are maintained under '/dev/disk'
# See man pages fstab(5), findfs(8), mount(8) and/or blkid(8) for more info
#
/dev/mapper/centos-root /                       xfs     defaults        0 0
UUID=24ced1f3-23fa-4e79-8929-9a769665c255 /boot                   xfs     defaults        0 0
/dev/mapper/centos-swap swap                    swap    defaults        0 0


**rc.local:**

#!/bin/bash
# THIS FILE IS ADDED FOR COMPATIBILITY PURPOSES
#
# It is highly advisable to create own systemd services or udev rules
# to run scripts during boot instead of using this file.
#
# In contrast to previous versions due to parallel execution during boot
# this script will NOT be run after all other services.
#
# Please note that you must run 'chmod +x /etc/rc.d/rc.local' to ensure
# that this script will be executed during boot.

touch /var/lock/subsys/local



thanks

**Comment by historicbruno** [comment]
I have the same problem with Nagios 4.1.1 on CentOS 7.2.1511, I followed the install guide I have used in the past (with some modifications): https://assets.nagios.com/downloads/nagioscore/docs/Installing_Nagios_Core_From_Source.pdf

I also have NRPE 3.0, pnp4nagios, and mk-livestatus 1.2.8p1 installed.
I disabled the NRPE service for testing, that had no effect, so I think it can be ruled out as a cause.

The entire contents of the var/ folder disappear on my system. Is Nagios _supposed_ to wipe out and recreate the var/ folder on reboot, or is this buggy behavior?

**Comment by box293** [comment]
The var folder should not be wiped as this is where the following long-term files live:

nagios.log
archives/
retention.dat

On a clean test system I have here with just Core 4.2.0 on CentOS Linux release 7.2.1511 the folder is not wiped on a reboot. I created a test file in the /usr/local/nagios/var/ folder and after the OS rebooted the file was still there. I also created separate test files in the rw and spool folders and they were still there after a reboot.

Does this problem only occur on an Operating System reboot? What happens if you stop the nagios service, does the folder still exist?

**Comment by historicbruno** [comment]
For me, there is no problem restarting the service, once the folders are manually recreated. It is only on reboot that the var/ folder is wiped out, even new files I created there for testing.

**Comment by historicbruno** [comment]
Also worth noting is that I have SELinux disabled.

**Comment by amitwein** [comment]
same here. no problem stopping or restarting Nagios service.
the folder is wiped out only after reboot.

**Comment by amitwein** [comment]
I have a workaround.
I have changed in nagios.cfg file, the location of:
nagios.cmd , nagios.qh, live files to:  /usr/local/nagios/var instead of /usr/local/nagios/var/rw 
also. changed the configuration of xinetd livestatus file to point to /usr/local/nagios/var.
now, after reboot, nagios is up and running.

can somebody point out if any problem may occur according to this configuration?

**Comment by jfrickson** [comment]
Cent 7 uses `systemd`. See if you have `ndo2db` and/or `nrpe` in both `/usr/lib/systemd/system` and in `/etc/init.d`. If so, make sure the one in `/etc/init.d` is disabled and deleted or renamed.

I doubt this is the reason, but it's worth ruling out.

**Comment by stknohg** [comment]
Hi, I had the same problem on 

* CentOS 7.2.1511 on ESXi
* Nagios Core 4.1.1
* NRPE 3.0

In my case, the cause was `/usr/lib/tmpfile.d/nrpe.conf`.  

`nrpe.conf` configuration is below. 

```sh
#Type	Path			Mode	UID				GID				Age	Argument
D		/usr/local/nagios/var	0755	[install user]	[install group]	-	-
```

I removed this file(I only use NRPE as plugin), and soleved the problem.

**Comment by jfrickson** [comment]
Ah, ok. That was to make sure the PID directory existed. Which on the default install messes up `/usr/local/nagios/var`. But if the PID dir is `/var/run/nrpe` it's needed. I think the `Type` field needs to be a lower-case `d` instead. I'll make that change.

**Comment by jfrickson** [comment]
Fixed in nrpe 'maint' branch via commit https://github.com/NagiosEnterprises/nrpe/commit/83137533bbaebd4346eb8e387ed92490afaf5199
Fix will be in the NRPE 3.0.1 release coming up shortly.
